search/geocoder.cpp

```diff
@@ -1573,7 +1573,9 @@ void Geocoder::MatchPOIsAndBuildings(BaseContext & ctx, size_t curToken, CBV con
           continue;
       }
       else if (layer.m_sortedFeatures->empty() ||
-               house_numbers::LooksLikeHouseNumberStrict(layer.m_subQuery))
+              /// @todo The crutch, but can't invent a better solution now. Should refactor layers iteration.
```

Comment:
Am I correctly understanding that any number in the query now counts as a house number?

Reply:
"... house number _only_"? No.
In your example searching for "21":
- before - no "Forever 21" results
- after - "Forever 21" should present